hcmanager/utils_test.go

```diff
@@ -29,80 +30,44 @@ import (
-		_, _, err := hcManager.FetchBmhHostList("sample")
-		if err != nil {
-			Expect(err).To(HaveOccurred())
-		}
-
-	})
-
-	It("Should return error for empty hardware profile", func() {
-		err := hcManager.ValidateExtractedHardwareProfile(getEmptyProfile())
-		if err != nil {
-			Expect(err).To(HaveOccurred())
-		}
-	})
-
-	It("Should validate extracted hardware profile", func() {
-		result := hcManager.ValidateExtractedHardwareProfile(getExtractedHardwareProfile())
-		if result == nil {
-			Expect(result).Should(BeNil())
-		}
-	})
-
-	It("Should return error for invalid CPU details in hardware profile", func() {
-		err := hcManager.ValidateExtractedHardwareProfile(getInvalidCPUProfile())
-		if err != nil {
-			Expect(err).To(HaveOccurred())
-		}
-	})
-
-	It("Should return error for invalid DISK details in hardware profile", func() {
-		err := hcManager.ValidateExtractedHardwareProfile(getInvalidDiskProfile())
-		if err != nil {
-			Expect(err).To(HaveOccurred())
-		}
-	})
-
-	It("Should return error for invalid NIC details in hardware profile", func() {
-		hardwareClassification := &hwcc.HardwareClassification{}
-		err := hcManager.ValidateExtractedHardwareProfile(getInvalidNicProfile())
-		SetStatus(hardwareClassification, hwcc.ProfileMatchStatusEmpty, hwcc.ProfileMisConfigured, err.Error())
-		if err != nil {
-			Expect(hardwareClassification.Status.ErrorType).Should(Equal(hwcc.ProfileMisConfigured))
-			Expect(err).To(HaveOccurred())
-		}
-	})
-
-	It("Should return error for invalid RAM details in hardware profile", func() {
-		err := hcManager.ValidateExtractedHardwareProfile(getInvalidRAMProfile())
-		if err != nil {
-			Expect(err).To(HaveOccurred())
-		}
-	})
-
-	It("Should return error for missing NIC details in hardware profile", func() {
-		err := hcManager.ValidateExtractedHardwareProfile(getMissingNicDetails())
-		if err != nil {
-			Expect(err).To(HaveOccurred())
-		}
-	})
+	DescribeTable("Test fetch BaremetalHost list",
+		func(namespace string) {
+			hostTest := getHosts()
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), hostTest...)
+			hcManager := NewHardwareClassificationManager(c, klogr.New())
+			result, _, err := hcManager.FetchBmhHostList(namespace)
+			if err != nil {
+				Expect(len(result)).To(BeZero())
+			}
+			if len(result) > 0 {
+				Expect(len(result)).Should(Equal(2))
+			}
+		},
+		Entry("Should fetch BaremetalHosts in ready state and under metal3 namespace", getNamespace()),
+		Entry("Should return error while fetching BaremetalHosts", "sample"),
+	)
+
+	DescribeTable("Test user profiles",
+		func(hwcc hwcc.HardwareCharacteristics) {
+			hostTest := getHosts()
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), hostTest...)
+			hcManager := NewHardwareClassificationManager(c, klogr.New())
+			err := hcManager.ValidateExtractedHardwareProfile(hwcc)
+			if err != nil {
+				Expect(err).To(HaveOccurred())
+			} else {
+				Expect(err).Should(BeNil())
+			}
```

Comment:
this does not test anything. Please check my previous suggestion.

Reply:
done. @maelk @digambar15 kindly review